File: backend/src/main/java/com/unidadeducativa/academia/asignaciondocente/model/AsignacionDocente.java
package com.unidadeducativa.academia.asignaciondocente.model;

import com.unidadeducativa.academia.curso.model.Curso;
import com.unidadeducativa.academia.materia.model.Materia;
import com.unidadeducativa.personas.profesor.model.Profesor;
import com.unidadeducativa.academia.gestion.model.GestionAcademica;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "asignacion_docente", uniqueConstraints = @UniqueConstraint(columnNames = { "id_profesor", "id_materia",
                "id_curso", "id_gestion" }))
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AsignacionDocente {

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(name = "id_asignacion")
        private Long idAsignacion;

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "id_profesor", nullable = false)
        private Profesor profesor;

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "id_materia", nullable = false)
        private Materia materia;

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "id_curso", nullable = false)
        private Curso curso;

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "id_gestion", nullable = false)
        private GestionAcademica gestion;

        @Column(nullable = false)
        @Builder.Default
        private Boolean estado = true;

        @Column(name = "fecha_inicio", nullable = false)
        @Builder.Default
        private LocalDate fechaInicio = LocalDate.now();

        @Column(name = "fecha_fin")
        private LocalDate fechaFin;
}

File: frontend/lib/features/director/models/asignacion_docente.dart
class AsignacionDocente {
  final int idAsignacion;
  final int idProfesor;
  final String nombreProfesor;
  final int idMateria;
  final String nombreMateria;
  final int idCurso;
  final String nombreCurso;
  final int idGestion;
  final int anioGestion;
  final bool estado;

  AsignacionDocente({
    required this.idAsignacion,
    required this.idProfesor,
    required this.nombreProfesor,
    required this.idMateria,
    required this.nombreMateria,
    required this.idCurso,
    required this.nombreCurso,
    required this.idGestion,
    required this.anioGestion,
    required this.estado,
  });

  factory AsignacionDocente.fromJson(Map<String, dynamic> json) {
    return AsignacionDocente(
      idAsignacion: json['idAsignacion'],
      idProfesor: json['idProfesor'],
      nombreProfesor: json['nombreProfesor'] ?? 'Sin nombre',
      idMateria: json['idMateria'],
      nombreMateria: json['nombreMateria'] ?? 'Sin materia',
      idCurso: json['idCurso'],
      nombreCurso: json['nombreCurso'] ?? '',
      idGestion: json['idGestion'] ?? 0,
      anioGestion: json['anioGestion'] ?? 0,
      estado: json['estado'] ?? true,
    );
  }
}
